apps/src/code-studio/components/ToggleSwitch.jsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import Radium from 'radium';
+import PropTypes from 'prop-types';
+import color from '@cdo/apps/util/color';
+import './styles.scss';
+
+// An accessible toggle, adapted from https://kittygiraudel.com/2021/04/05/an-accessible-toggle/
+class ToggleSwitch extends React.Component {
+  static propTypes = {
+    isToggledOn: PropTypes.bool.isRequired,
+    onToggle: PropTypes.func.isRequired,
+    label: PropTypes.string.isRequired
+  };
+
+  render() {
+    const {isToggledOn, onToggle, label} = this.props;
+
+    let dynamicStyle = isToggledOn
+      ? styles.toggledOnIcon
+      : styles.toggledOffIcon;
+    const iconStyle = {...styles.icon, ...dynamicStyle};
+    return (
+      <button
+        ref={ref => (this.buttonDiv = ref)}
```

Comment:
What is the `ref` for?

Reply:
That was there because Radium requires a key or ref if you have nested elements that both use browser state modifiers, I ended up removing the modifier on the nested element so I'll remove those refs.